src/inject.js

```diff
@@ -7,35 +7,27 @@ import { MobXProviderContext } from "./Provider"
  * Store Injection
  */
 function createStoreInjector(grabStoresFn, component, injectNames, makeReactive) {
-    let displayName = getInjectName(component, injectNames)
-
-    class Injector extends Component {
-        static contextType = MobXProviderContext
-
-        render() {
-            const { forwardRef, ...props } = this.props
+    // Support forward refs
+    let Injector = React.forwardRef((props, ref) => {
+        const newProps = { ...props }
+        const context = React.useContext(MobXProviderContext)
+        Object.assign(newProps, grabStoresFn(context || {}, newProps) || {})
 
-            Object.assign(props, grabStoresFn(this.context || {}, props) || {})
+        if (ref && !isStateless(component)) {
```

Comment:
What about moving this condition outside and basically avoid calling `forwardRef` for a non-stateless component?

Although I do wonder what's the purpose of that. Why is the ref passed to class components only? What am I missing here?

Reply:
> What about moving this condition outside and basically avoid calling forwardRef for a non-stateless component?

I will try it.

> Although I do wonder what's the purpose of that. Why is the ref passed to class components only? What am I missing here?

Because function components cannot be given refs?

Reply:
> Because function components cannot be given refs?

True.

Btw, might as well rename `isStateless` to `isFunctional` to stay current :)

Reply:
> What about moving this condition outside and basically avoid calling forwardRef for a non-stateless component?

I’m having trouble trying to figure out how to do it in a simple way without repetition.
https://gist.github.com/vkrol/e2804b367c28f96b0802f21fa1845c18